src/System.Management.Automation/DscSupport/JsonCimDSCParser.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Management.Automation;
+using System.Security;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Internal.Json
+{
+    /// <summary>
+    /// Class that does high level Cim schema parsing.
+    /// </summary>
+    internal class CimDSCParser
+    {
+        private readonly JsonDeserializer _json_deserializer;
+        
+        internal CimDSCParser()
+        {
+            _json_deserializer = JsonDeserializer.Create();
+        }
+
+        internal IEnumerable<PSObject> ParseSchemaJson(string filePath, bool useNewRunspace = false)
+        {
+            try
+            {
+                string json = File.ReadAllText(filePath);
+                string fileNameDefiningClass = Path.GetFileNameWithoutExtension(filePath);
+                int dotIndex = fileNameDefiningClass.IndexOf(".schema", StringComparison.InvariantCultureIgnoreCase);
+                if (dotIndex != -1)
+                {
+                    fileNameDefiningClass = fileNameDefiningClass.Substring(0, dotIndex);
+                }
+
+                var result = _json_deserializer.DeserializeClasses(json, useNewRunspace);
+                foreach (dynamic classObject in result)
```

Comment:
I think using `PSObject` is clearer and more performant than using `dynamic` here

Reply:
removing dynamic gives compilation errors like `'PSObject' does not contain a definition for 'SuperClassName' and no accessible extension method...`
This was suggested by @daxian-dbw .